src/useradd.c

```diff
@@ -764,11 +765,7 @@ static int get_groups (char *list)
 	/*
 	 * Free previous group list before creating a new one.
 	 */
-	int i = 0;
-	while (NULL != user_groups[i]) {
-		free(user_groups[i]);
-		user_groups[i++] = NULL;
-	}
+	free_groups ();
```

Comment:
Please do this refactor in a preparation patch, so that the commit that adds the feature is minimal.